webserver/src/main/java/io/kestra/webserver/controllers/ExecutionController.java

```diff
@@ -962,4 +964,43 @@ public Flowable<Event<Execution>> follow(
                 }
             });
     }
+
+    @ExecuteOn(TaskExecutors.IO)
+    @Get(uri = "executions/{executionId}/file/preview", produces = MediaType.TEXT_JSON)
+    @Operation(tags = {"Executions"}, summary = "Get file preview for an execution")
+    public HttpResponse<?> filePreview(
+        @Parameter(description = "The execution id") @PathVariable String executionId,
+        @Parameter(description = "The internal storage uri") @QueryValue URI path
+    ) throws IOException {
+        HttpResponse<StreamedFile> httpResponse = this.validateFile(executionId, path, "/api/v1/executions/{executionId}/file?path=" + path);
+        if (httpResponse != null) {
+            return httpResponse;
+        }
+
+        InputStream fileStream = storageInterface.get(path);
+
+        String extension = FilenameUtils.getExtension(path.toString());
+
+        if (ImageFileExtension.isImageFileExtension(extension)) {
+            byte[] imageBytes = IOUtils.toByteArray(fileStream);
+            String base64Image = Base64.getEncoder().encodeToString(imageBytes);
+
+            return HttpResponse.ok(PreviewResponse.builder().extension(extension).content(base64Image).build());
+        }
+        BufferedReader reader = new BufferedReader(new InputStreamReader(fileStream));
+        String line = reader.readLine();
+        int lineCount = 0;
+
+        StringBuilder contentBuilder = new StringBuilder();
+
+        while (line != null && lineCount < 100) {
```

Comment:
Not all files are multi-line, you can have a huge file with a one-in JSON in it.
I'm not sure we can do something better than that anyway

Reply:
One solution would be to limit the size of the content return, limit to a size of 2mb for example so even if my file is a heavy one-line, I will return limited content

Reply:
Agree that we need an limit in MB or number of lines, the first that will cut of